### DISCUSSION TITLE
Encrypt and decrypt pb authStore

### DISCUSSION THREAD

**Discussion by Graciasjr** [pocketbase/pocketbase#4655] [Q&A]
Hi.
I would like to know if we could encrypt the pb_authstore data which is stored in the browser and decrypt it upon recovery?

**Comment by ganigeorgiev** [comment]
Please don't just "prompt" and try to elaborate as much as possible to understand the bigger picture of your problem.

Without context on what you are trying to do and why, the question doesn't make sense to me.

The JS SDK `pb.authStore` is using the browser LocalStorage by default for persistence. If you want to create your own custom store you can check the [Custom auth store example](https://github.com/pocketbase/js-sdk#custom-auth-store).

**Reply by Graciasjr** [reply]
Please forgive me for not providing enough context.
In fact what I'm trying to do is not to customize the authstore but rather to find a way to encrypt the data that goes into the localstorage with an external library.
Because, the goal is to make the data that goes from the localstorage unusable

**Reply by ganigeorgiev** [reply]
It doesn't explain what is the actual problem you are trying to solve, nor why you want to do it.
Again, without context encrypting the browser localStorage content doesn't make sense to me.
Where the encryption key will be stored? If your goal is to "hide" the stored token from the user what prevents the user for sending a direct http request against the API auth endpoint and retrieve it? Or you want the auth response to be encrypted on server-side level? If that's the case - why? What you are trying to protect/"hide"?

**Comment by gokulgseven** [comment]
encrypting the auth store doesnt make sense. if you want to make the application secure -> use cookies (cookies are domain specific and wont work on some other device )-> you will find a discussion on cookies somewhere in the discussion page-> it should detail how to get it working (along with it use refresh tokens and make the token time period small, thats all you can do, no application is completely secure 

even if you do manage to encrypt the token , the code to encrypt it will be stored on the browser (along with whatever encryption key you have used, any experienced user can just to see the code and recreate the token) , also anyone can just see the actual token being sent with any of the requests in the console. 

i had a similar use case once, where i just wanted to stop inexperienced user (following some youtube video ) to call the apis manually
what i did was just added 1 random characters in the middle of the token before saving it, and before every call just removed that random character. (wont stop a motivated user from calling the api, but will stop  a couple of them users ) (this was a long while back -> dont use this method)

you can still see the code and reverse engineer it , you can see the actual api call that the app is making and get the correct token from there ,
